### PR TITLE
feat: Add active element support

### PR DESCRIPTION
**Pull request by vane**
Add support to retrieve active element from android application.
Corresponding PR.
https://github.com/appium/appium-uiautomator2-server/pull/423

### CODE REVIEW

File: lib/commands/general.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ let extensions = {},
     commands = {},
     helpers = {};
 
+commands.active = async function () {
```
Comment (jlipps) [inline lib/commands/general.js:12]: is there an element commands file? doesn't _really_ matter where this goes but if there's an elements file that would be more sensible I think.

**Comment by mykola-mokhnach** [comment]
Please bump the minimum appium-uiautomator2-server component version in package.json (should be `^4.22.0`)

File: lib/commands/general.js [hunk 2]
```diff
@@ -9,6 +9,10 @@ let extensions = {},
...
+  return await this.uiautomator2.jwproxy.command('/element/active', 'POST', {});
```
Comment (mykola-mokhnach) [inline lib/commands/general.js:13]: the request method must be GET according to the W3C protocol: https://www.w3.org/TR/webdriver/#dfn-get-active-element

File: lib/commands/general.js (hunk 1 above)
Comment (vane) [reply lib/commands/general.js:12]: Thanks for feedback. I put it to `general.js` to be consistent with `appium-ios-driver` project but I can move it to element.

File: lib/commands/general.js (hunk 2 above)
Comment (vane) [reply lib/commands/general.js:13]: hmm there is call for active element in `appium-xcuitest-driver` that is using `POST` method I wanted to get it consistent with that method https://github.com/appium/appium-xcuitest-driver/blob/70adbf60bbf66deb86388e602b53eae902a3e04f/lib/driver.js#L100

Comment (jzaleski) [reply lib/commands/general.js:13]: Per the comment on https://github.com/appium/appium-uiautomator2-server/pull/424 we'll submit a PR to deprecate the `POST` of `/active` if it still exists on _XCUITest_ and add the `GET` variant.

File: lib/commands/general.js (hunk 1 above)
Comment (vane) [reply lib/commands/general.js]: Ok I moved it to `element.js` and changed to GET.

**Comment by jzaleski** [comment]
I believe this: https://github.com/appium/appium-uiautomator2-server/pull/424 will mandate another server-version bump so per @mykola-mokhnach's comment above please set the minimum-version in the package.json to whatever that is (presumably `4.22.1`).

**Comment by jzaleski** [comment]
I would assume the validation checks will pass once https://github.com/appium/appium-uiautomator2-server/pull/424 is merged as right now version `4.22.1` of the _Server_ component does not exist.

File: lib/commands/element.js [hunk 3]
```diff
@@ -9,6 +9,10 @@ function toBool (s) {
   return _.isString(s) ? (s.toLowerCase() === 'true') : !!s;
 }
 
+commands.active = async function () {
```
Comment (mykola-mokhnach) [inline lib/commands/element.js:12]: please add a name to the anonymous function - it makes easier to debug the the stuff:

`async function ()` -> `async function active ()`

**Comment by mykola-mokhnach** [comment]
The patch to the server package with version 4.22.1 has been published

File: lib/commands/element.js (hunk 3 above)
Comment (vane) [reply lib/commands/element.js]: Right, fixed. Sorry for trouble with w3c stuff.

Comment (mykola-mokhnach) [reply lib/commands/element.js]: np, that is why we have the review process :)

**Review by mykola-mokhnach — APPROVED** [review]

File: lib/commands/element.js [hunk 4]
```diff
@@ -9,6 +9,10 @@ function toBool (s) {
   return _.isString(s) ? (s.toLowerCase() === 'true') : !!s;
 }
 
+commands.active = async function active () {
+  return await this.uiautomator2.jwproxy.command('/element/active', 'GET', {});
```
Comment (mykola-mokhnach) [inline lib/commands/element.js:13]: the last argument is redundant here since GET requests don't have bodies

**Comment by mykola-mokhnach** [comment]
@vane Would it be possible to add a simple end-to-end test for this feature?

File: lib/commands/element.js (hunk 4 above)
Comment (vane) [reply lib/commands/element.js]: removed

**Comment by vane** [comment]
Added simple e2e test as requested

**Comment by jlipps** [comment]
awesome work, thanks @vane !

**Pull request merged**